projects/storefrontlib/src/lib/product/components/product-details/product-summary/product-summary.component.ts

```diff
@@ -44,7 +45,83 @@ export class ProductSummaryComponent {
     );
   }
 
-  launchReview() {
-    this.openReview.emit();
+  // Checks if product has reviews
+  // hasReviews(): Observable<boolean> {
+  //   return new Observable((observer: Observer<boolean>) => {
+  //     if (this.getReviewsComponent()) {
+  //       // Product reviews have rendered
+  //       observer.next(true);
+  //     }
+
+  //     observer.complete();
+  //   });
+  // }
+
+  // NOTE: Does not currently exists as its own component
+  // but part of tabs component. This is likely to change in refactor.
+  private getReviewsComponent(): Element {
+    return document.querySelector('cx-product-reviews');
+  }
+
+  // Get Tabs Component if exists on page
+  private getTabsComponent(): Element {
+    return document.querySelector('cx-product-tabs');
+  }
+
+  // Get Reviews Tab if exists on page
+  private getReviewsTab(): HTMLElement {
+    if (this.getTabsComponent()) {
+      // NOTE: Reads through h3 tags to click on correct tab
+      // There may be a better way of doing this now/after refactor
+      const h3Elements: HTMLCollectionOf<
+        HTMLElement
+      > = this.getTabsComponent().getElementsByTagName('h3');
+
+      // Use translated label for Reviews tab reference
+      const reviewsTabTitle = this.translatePipe.transform(
+        'productDetails.label.reviews'
+      );
+
+      // Look through h3 tab elements until finding tab with Reviews reference
+      for (const h3Element of Array.from(h3Elements)) {
+        if (h3Element.innerHTML.indexOf(reviewsTabTitle) > -1) {
+          return h3Element;
+        }
+      }
+    }
+
+    return null;
+  }
+
+  // Scroll to views component on page and click "Reviews" tab
+  showReviews() {
```

Comment:
I don't think so we need to attend negative situation (we can't click button if it doesn't exist). Console logs are undesirable, but if you like lets import { ServerConfig } from '@spartacus/core';

then code can looks like:
```ts
const reviewsTab = this.getReviewsTab();
const reviewsComponent = this.getReviewsComponent();

if (reviewsTab && reviewsComponent) {
  reviewsComponent.scrollIntoView();
  this.clickTabIfInactive(reviewsTab);
} else {
  this.warn(`Cannot find tabs or Reviews component');
}

////

clickTabIfInactive(tab: HTMLElement): void {
  const reviewsTab = this.getReviewsTab();
  if (
    !tab.classList.contains('active') ||
    tab.classList.contains('toggled')
  ) {
    tab.click();
  }
}

private warn(text: string): void { // In the future it will be moved to logger service/global function
  if (!this.config.production) {
    console.warn(text)
  }
}
```

Reply:
Ah you're right. Those checks are redundant now.